official/resnet/ctl/ctl_imagenet_main.py

```diff
@@ -127,50 +102,87 @@ def run(flags_obj):
       is_training=True,
       data_dir=flags_obj.data_dir,
       batch_size=flags_obj.batch_size,
-      num_epochs=flags_obj.train_epochs,
       parse_record_fn=parse_record_keras,
       datasets_num_private_threads=flags_obj.datasets_num_private_threads,
       dtype=dtype)
 
+  if strategy:
+    train_ds = strategy.experimental_distribute_dataset(train_ds)
+  
   test_ds = None
   if not flags_obj.skip_eval:
     test_ds = input_fn(
         is_training=False,
         data_dir=flags_obj.data_dir,
         batch_size=flags_obj.batch_size,
-        num_epochs=flags_obj.train_epochs,
         parse_record_fn=parse_record_keras,
         dtype=dtype)
 
     if strategy:
       test_ds = strategy.experimental_distribute_dataset(test_ds)
-      steps_per_eval = IMAGENET_VALIDATION_IMAGES // flags_obj.batch_size
 
-  if strategy:
-    train_ds = strategy.experimental_distribute_dataset(train_ds)
-  steps_per_epoch = APPROX_IMAGENET_TRAINING_IMAGES // flags_obj.batch_size
+  return train_ds, test_ds
+
+
+def get_num_train_iterations(flags_obj):
+  steps_per_epoch = imagenet_main.NUM_IMAGES['train'] // flags_obj.batch_size
   train_epochs = flags_obj.train_epochs
 
   if flags_obj.train_steps:
     train_steps = min(flags_obj.train_steps, steps_per_epoch)
     train_epochs = 1
 
+  steps_per_eval = imagenet_main.NUM_IMAGES['validation'] // flags_obj.batch_size
```

Comment:
`steps_per_eval` name seems a bit confusing to me, perhaps just be `eval_steps`. 
steps_per_eval is sometimes used to describe how many training steps to do between 2 evals..

Reply:
Done.